File: tests/output/jpa/src/main/javagen/restaurant/jpa_feign/dtos/restaurant/PlatItem.java
////
//// ATTENTION CE FICHIER EST GENERE AUTOMATIQUEMENT !
////

package restaurant.jpa_feign.dtos.restaurant;

import java.math.BigDecimal;

import jakarta.annotation.Generated;

import restaurant.jpa_feign.enums.restaurant.CategoriePlatCode;

/**
 * Détail d'un plat en liste.
 */
@Generated("TopModel : https://github.com/klee-contrib/topmodel")
public interface PlatItem {

	/**
	 * Identifiant du plat.
	 */
	Integer getId();

	/**
	 * Nom du plat.
	 */
	String getNom();

	/**
	 * Prix du plat.
	 */
	BigDecimal getPrix();

	/**
	 * Indique si le plat est disponible.
	 */
	Boolean getDisponible();

	/**
	 * Catégorie du plat.
	 */
	CategoriePlatCode getCategoriePlatCode();

	/**
	 * Identifiant du plat.
	 * @param id value to set.
	 */
	void setId(Integer id);

	/**
	 * Nom du plat.
	 * @param nom value to set.
	 */
	void setNom(String nom);

	/**
	 * Prix du plat.
	 * @param prix value to set.
	 */
	void setPrix(BigDecimal prix);

	/**
	 * Indique si le plat est disponible.
	 * @param disponible value to set.
	 */
	void setDisponible(Boolean disponible);

	/**
	 * Catégorie du plat.
	 * @param categoriePlatCode value to set.
	 */
	void setCategoriePlatCode(CategoriePlatCode categoriePlatCode);
}

File: tests/output/vanilla-js/model/restaurant/plat-item.ts
////
//// ATTENTION CE FICHIER EST GENERE AUTOMATIQUEMENT !
////

import {DO_BOOLEEN, DO_CODE, DO_LIBELLE, DO_PRIX, DO_SEQ_ID} from "../../domains";

import {CategoriePlatCode} from "./enums";

export interface PlatItem {
    id: number;
    nom: string;
    prix: number;
    disponible: boolean;
    categoriePlatCode: CategoriePlatCode;
}

export const PlatItemEntity = {
    id: {
        type: "field",
        name: "id",
        domain: DO_SEQ_ID,
        isRequired: true,
        label: "Id"
    },
    nom: {
        type: "field",
        name: "nom",
        domain: DO_LIBELLE,
        isRequired: true,
        label: "Nom"
    },
    prix: {
        type: "field",
        name: "prix",
        domain: DO_PRIX,
        isRequired: true,
        label: "Prix"
    },
    disponible: {
        type: "field",
        name: "disponible",
        domain: DO_BOOLEEN,
        defaultValue: true,
        isRequired: true,
        label: "Disponible"
    },
    categoriePlatCode: {
        type: "field",
        name: "categoriePlatCode",
        domain: DO_CODE,
        isRequired: true,
        label: "CategoriePlatCode"
    }
} as const;
